apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnection.cs

```diff
@@ -798,34 +798,40 @@ public void Dispose()
 #if NET45 || NET451
 
         /// <summary>
-        /// Opens and establishes a session with the Spanner database with the given
-        /// <see cref="TimestampBound" /> settings that control the implicit transaction created
-        /// from the active <see cref="System.Transactions.TransactionScope" />.
+        /// If this connection is being opened within a <see cref="System.Transactions.TransactionScope" />, this forces
+        /// the created Cloud Spanner transaction to be a readonly transaction with the given
+        /// <see cref="TimestampBound" /> settings.
         /// </summary>
         /// <param name="timestampBound">
         /// Specifies the timestamp or maximum staleness of a
         /// read operation. Must not be null.
         /// </param>
-        public void OpenWithTimeBoundSettings(TimestampBound timestampBound)
+        public void OpenAsReadOnly(TimestampBound timestampBound)
         {
             _timestampBound = timestampBound;
             Open();
+            if (_volatileResourceManager == null) {
+                throw new InvalidOperationException("OpenAsReadOnly should only be called within a TransactionScope.");
```

Comment:
Can this only be checked *after* calling `Open`? It's a bit odd to check that state is valid *after* making changes. But this is probably me not understanding how `_volatileResourceManager` works.